File: lib/models/fbc.dart
class FBC {
  final int id;
  final int userId;
  final DateTime testDate;
  final double haemoglobin;
  final double totalLeucocyteCount;
  final double plateletCount;
  final String? imageUrl;

  FBC({
    required this.id,
    required this.userId,
    required this.testDate,
    required this.haemoglobin,
    required this.totalLeucocyteCount,
    required this.plateletCount,
    this.imageUrl,
  });

  // Factory constructor to create an FBC instance from a map (JSON)
  factory FBC.fromJson(Map<String, dynamic> json) {
    return FBC(
      id: json['id'] as int,
      userId: json['user_id'] as int,
      testDate: DateTime.parse(json['test_date'] as String),
      haemoglobin: json['haemoglobin'] as double,
      totalLeucocyteCount: json['total_leucocyte_count'] as double,
      plateletCount: json['platelet_count'] as double,
      imageUrl: json['image_url'] as String?,
    );
  }

  // Method to convert an FBC instance to a map (JSON)
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'user_id': userId,
      'test_date': testDate.toIso8601String(),
      'haemoglobin': haemoglobin,
      'total_leucocyte_count': totalLeucocyteCount,
      'platelet_count': plateletCount,
      'image_url': imageUrl,
    };
  }
}

File: models-kotlin/FBC.kt
package com.healthtracker.models

import com.google.gson.annotations.SerializedName
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

data class FBC(
    val id: Int,
    @SerializedName("user_id")
    val userId: Int,
    @SerializedName("test_date")
    val testDate: LocalDateTime,
    val haemoglobin: Double,
    @SerializedName("total_leucocyte_count")
    val totalLeucocyteCount: Double,
    @SerializedName("platelet_count")
    val plateletCount: Double,
    @SerializedName("image_url")
    val imageUrl: String? = null
) {
    companion object {
        private val dateFormatter = DateTimeFormatter.ISO_LOCAL_DATE_TIME

        fun fromJson(json: Map<String, Any?>): FBC {
            return FBC(
                id = json["id"] as Int,
                userId = json["user_id"] as Int,
                testDate = LocalDateTime.parse(json["test_date"] as String, dateFormatter),
                haemoglobin = json["haemoglobin"] as Double,
                totalLeucocyteCount = json["total_leucocyte_count"] as Double,
                plateletCount = json["platelet_count"] as Double,
                imageUrl = json["image_url"] as String?
            )
        }
    }

    fun toJson(): Map<String, Any?> {
        return mapOf(
            "id" to id,
            "user_id" to userId,
            "test_date" to testDate.format(dateFormatter),
            "haemoglobin" to haemoglobin,
            "total_leucocyte_count" to totalLeucocyteCount,
            "platelet_count" to plateletCount,
            "image_url" to imageUrl
        )
    }
}